app/src/main/java/de/othr/plantico/ui/plantDetails/PlantDetailActivity.kt

```diff
@@ -0,0 +1,151 @@
+package de.othr.plantico.ui.plantDetails
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.android.flexbox.FlexDirection
+import com.google.android.flexbox.FlexboxLayoutManager
+import com.google.android.flexbox.JustifyContent
+import de.othr.plantico.PlantViewModel
+import de.othr.plantico.PlantViewModelFactory
+import de.othr.plantico.R
+import de.othr.plantico.database.PlantApplication
+import de.othr.plantico.database.entities.Plant
+import de.othr.plantico.databinding.ActivityHomescreenBinding
+import de.othr.plantico.databinding.ActivityPlantDetailPageBinding
+import de.othr.plantico.ui.PlantActivity
+import de.othr.plantico.ui.homescreen.HomescreenActivity
+import de.othr.plantico.ui.homescreen.PlantAdapter
+import de.othr.plantico.ui.homescreen.WateringAdapter
+import de.othr.plantico.ui.search.SearchableActivity
+
+class PlantDetailActivity() : AppCompatActivity() {
+
+
+    private var plantID = 0
+    private val plantList = ArrayList<Plant>()
+    private lateinit var binding: ActivityPlantDetailPageBinding
+    private val plantViewModel: PlantViewModel by viewModels {
+        PlantViewModelFactory((application as PlantApplication).repository)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityPlantDetailPageBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+
+        val mActionBarToolbar = binding.toolbarPlantView
+        mActionBarToolbar.title = ""
+        setSupportActionBar(mActionBarToolbar)
+
+        val bundle = savedInstanceState ?: intent.extras
+        if (bundle != null) {
+            plantID = bundle.getInt(PlantActivity.SELECTED_PLANT)
+
+        }
+
+        binding.categoriesItemList.setHasFixedSize(true)
+        val layout = FlexboxLayoutManager(
+            this
+        )
+        layout.flexDirection = FlexDirection.ROW
+        layout.justifyContent = JustifyContent.FLEX_START
+        binding.categoriesItemList.layoutManager = layout
+        val adapter = CategorieListAdapter(
+            this
+        )
+        binding.categoriesItemList.adapter = adapter
+
+        var plant: Plant? = null
+        plantViewModel.allPlants.observe(this) { plants ->
+            // Update the cached copy of the words in the adapter.
+            plants.let {
+                plantList.clear()
+                plantList.addAll(it)
+                plant = plantList.find { plant -> plant.id == plantID }!!
+                if (plant != null) {
+
```

Comment:
Is there a reason why you don't use getPlantByID(plantID)?

Reply:
The problem is, that if I call viewModel.getPlantById(plantID) I get following error Message:

`java.lang.IllegalStateException: Cannot access database on the main thread since it may potentially lock the UI for a long period of time.`

This is because we access the database. There are two possible solutions we can set the database to allow querys on the MainThread or work with coroutines.

I personally check that when we allow it, it works. But this was something I wanted to discuss on Tuesday

Reply:
Oh I see now, I didn't implement it correctly, you're right. Then it's fine for now, I'll fix it in the coming days.